cloudformation.yaml

```diff
@@ -194,13 +194,13 @@ Resources:
                 - states:StartExecution
                 Resource: !Ref NodeRotationStepFunction
 
-  DiscoverAutoScalingGroupLambda:
```

Comment:
These CFN changes will remove/replace some lambda functions. Consequently I think we'll need to be a bit careful when rolling out this change as an in-progress execution of the Step Function might still be relying on some of the old lambdas for subsequent steps.

We could do this in two PRs:

1. Introduce all new lambdas (without deleting/replacing any old ones)
2. Delete old lambdas

Alternatively given the nature of this project, it's probably simplest to just ensure that these changes are deployed when the Step Function is not running.

Reply:
yeah, given there's only a few teams using this (one of which you belong to, and I one of the others, leaving perhaps only @rtyley 's team) best just to do it at a time outside of our scheduled executions, should be easy to coordinate. that said it's not quite ready to go as I need to get round to writing up the description properly...